bitmark-references/rpc-communication/rpc-communication.md

```diff
@@ -0,0 +1,117 @@
+# RPC
+
+## Introduction
+
+RPC is the abbreviation of remote procedure call, which is an inter-process communication technique used for client-server based applications. It is also known as subroutine call or function call.
+
+# Flow
+
+## Initialization
+
+When `bitmarkd` starts, it initialises RPC settings including maximum connection count, connection bandwidth, communication encryption key, binds RPC on specific port, listen RPC port.
+
+## Processing Messages
+
+Different RPC command is processed by differnet function, following sections describes each command with examples. Each message is in json format, with one key `method` denotes which remote method being executed, and the other key `params` denotes what arguments are passed.
+
+In order to test example, `bitmarkd` should be started in advance, default RPC port is on `2130`. `openssl` should be installed in advance to run communication.
+
+All methods are named in the format of `file_name.method_name`, for example, `Assets.Get` stands for method `Get` in file `assets.go`.
+
+Available commands:
+
+1. Assets.Get - retrieve assets information
+
+It is necessary to provide array of asset fingerprints to query asset information fingerprints).
+
```

Comment:
Many of these commands feel awkward in their description of the necessary inputs. I think we could make them less awkward with a simple and standardized parameter listing for each.

For example:

1. Assets.Get - retrieve assets information

Parameters:
* fingerprints: an array of asset fingerprints for querying asset information

Reply:
updated